README.md

```diff
@@ -1,23 +1,70 @@
 
 [![Build Status for the Core backend](https://github.com/devitocodes/devito/workflows/CI-core/badge.svg)](https://github.com/devitocodes/devito/actions?query=workflow%3ACI-core)
 [![Build Status with MPI](https://github.com/devitocodes/devito/workflows/CI-mpi/badge.svg)](https://github.com/devitocodes/devito/actions?query=workflow%3ACI-mpi)
 [![Code Coverage](https://codecov.io/gh/devitocodes/devito/branch/master/graph/badge.svg)](https://codecov.io/gh/devitocodes/devito)
+[![Slack Status](https://img.shields.io/badge/chat-on%20slack-%234A154B)](https://opesci-slackin.now.sh)
+
+[Devito](http://www.devitoproject.org) is a Python package to implement
+optimised tensor computation from high-level symbolic problem definitions.
+Devito builds on [SymPy](http://www.sympy.org/en/index.html) and employs
+automated code generation and just-in-time (JIT) compilation to execute
+optimized computational kernels on multiple computer platforms, including
+CPUs, GPUs, and clusters thereof.
+
+- [About Devito](#about-devito)
+- [Installation](#installation)
+- [Resources](#resources)
+- [Performance](#performance)
+- [Get in touch](#get-in-touch)
+
+## About Devito
+
+Devito provides a symbolic language to implement matrix-free operators, that is
+operators that do not require the explicit creation of a dense or sparse
+matrix. A typical use case is explicit finite difference (FD) methods for
+approximating partial differential equations. For example, a 2D diffusion
+operator may be implemented as follows
+
+```python
+>>> grid = Grid(shape=(10, 10))
+>>> f = TimeFunction(name='f', grid=grid, space_order=2)
+>>> eqn = Eq(f.dt, 0.5 * f.laplace)
+>>> op = Operator(Eq(f.forward, solve(eqn, f.forward)))
+```
 
-[Devito](http://www.devitoproject.org) is a software to
-implement optimised finite difference (FD) computation from
-high-level symbolic problem definitions. Starting from symbolic
-equations defined in [SymPy](http://www.sympy.org/en/index.html),
-Devito employs automated code generation and just-in-time (JIT)
-compilation to execute FD kernels on multiple computer platforms.
-
-## Get in touch
+An `Operator` generates low-level code from an ordered collection of `Eq`. This
+code may also be compiled and executed
 
-If you're using Devito, we would like to hear from you. Whether you
-are facing issues or just trying it out, join the
-[conversation](https://opesci-slackin.now.sh).
+```python
+>>> op(t=timesteps)
+```
 
-## Quickstart
+There is virtually no limit to the type of matrix-free operators that one can
+implement with Devito. The Devito compiler will automatically analyze the input,
+detect and apply optimizations (including single- and multi-node parallelism),
+and eventually generate code with arbitrarily complex loop nests, as required
+by the symbolic specification -- clearly, all this is hidden away to the user.
+
+Key features include:
+
+* A tensor language to express FD operators.
```

Comment:
IMO the language is still a functional symbolic language, the compiler is a tensorial compiler.

Reply:
Isn't a `Function` a tensor (since it's a discrete function)? `div` `curl` and all other operators make it a functional language, sure

Reply:
Its data is a tensor, but from a user/interface perspective, it is a pure functional object, you can do any sympy operation (integrals, polynomial, ...) on it. So yes on one side the compiler sees it as a tensor (nd-array) but the language used to define the object is still functional.

Reply:
OK, I think I see the confusion. You're right, functional is the right thing to use here